### PR TITLE
Switch to config file for pipeline parameters

### PR DESCRIPTION
**Pull request by ngreenwald**
This PR switches how most of the key variables for the pipeline are handled. It adds a configuration dictionary as part of the Context object. This dictionary holds the values for the parameters. 
1. For the majority of parameters, these must be supplied in the config file. However, to enable easy interactive use, I set it up so that a small subset can be modified directly from the `run()` call. 
2. I expanded test coverage for corner cases and made some fixes/tweaks for issues that came up along the way
3. I also moved the rest of the files out of the top-level Context file and into the `extras` dict in Context for better organization

### CODE REVIEW

File: example/test_config.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Structure data
```
[the remark this answers is not in the document]
Comment (ngreenwald) [reply example/test_config.toml]: If a variable is missing or set to blank, it goes to the default values present in the repo. If it has a value, that overrides the default

File: src/structure/structure_context.py [hunk 2]
```diff
@@ -41,6 +42,69 @@ def metrics_with_tag(tag: str) -> List[str]:
...
+    # structure parameters
+    vdw_radii: str = "ProtOr"
+    membrane_thickness: Optional[float] = 15
+
+    # mutagenesis data
+    mutation_data_path: Optional[Path] = None
+    mutation_data_chain: Optional[str] = None
+
+    # sequence features
+    aaindex_path: Path = 'data/aaindex_parsed_small.csv'
```
[the remark this answers is not in the document]
Comment (ngreenwald) [reply src/structure/structure_context.py]: This config file is for the entire pipeline, so it has default values for everything. If the user doesn't have any mutation data, it just calculates sequence features from the wildtype sequence.
